src/fruity/fruity-host-session.vala

```diff
@@ -299,7 +299,10 @@ namespace Frida {
 				var client = yield get_lockdown_client (cancellable);
 
 				try {
-					return yield client.start_service (service_name, cancellable);
+					if (service_name.length != 0)
+						return yield client.start_service (service_name, cancellable);
+					else
+						return client.stream;

```

Comment:
Here we have to create a brand new client and return its `stream`, because otherwise we risk that other API calls use the same shared client to start services while the application is using it for other things.

Reply:
ohh right, ok this needs also lifecycle management i guess, then.

Reply:
Lifecycle management isn't needed. Here we get a reference on the underlying `IOStream`, and right after the `LockdownClient` ref count will drop to zero, which means also its `PlistServiceClient` ref count will drop to zero, and the `IOStream` ref left will be held here. Because `LockdownClient` and `PlistServiceClient` only interact with the stream when asked to, and don't do anything to it during dispose/finalize, it is perfectly safe to hold on to the `IOStream` but let those other two objects die.